cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TopicOperator.java

```diff
@@ -289,10 +391,33 @@ String getAncillaryConfigMapKeyLogConfig() {
     }
 
     private List<Volume> getVolumes() {
-        return Collections.singletonList(createConfigMapVolume(logAndMetricsConfigVolumeName, ancillaryConfigName));
+        List<Volume> volumeList = new ArrayList<>();
+        volumeList.add(createConfigMapVolume(logAndMetricsConfigVolumeName, ancillaryConfigName));
+        volumeList.add(createSecretVolume(TLS_SIDECAR_VOLUME_NAME, TopicOperator.secretName(cluster)));
+        return volumeList;
     }
 
     private List<VolumeMount> getVolumeMounts() {
-        return Collections.singletonList(createVolumeMount(logAndMetricsConfigVolumeName, logAndMetricsConfigMountPath));
+        List<VolumeMount> volumeMountList = new ArrayList<>();
+        volumeMountList.add(createVolumeMount(logAndMetricsConfigVolumeName, logAndMetricsConfigMountPath));
+        volumeMountList.add(createVolumeMount(TLS_SIDECAR_VOLUME_NAME, TLS_SIDECAR_VOLUME_MOUNT));
+        return volumeMountList;
+    }
+
+    /**
+     * Generate the Secret containing CA self-signed certificates for internal communication
+     * It also contains the private key-certificate (signed by internal CA) for communicating with Zookeeper and Kafka
+     * @return The generated Secret
+     */
+    public Secret generateSecret() {
+        Map<String, String> data = new HashMap<>();
+        data.put("cluster-ca.crt", Base64.getEncoder().encodeToString(clusterCA.cert()));
```

Comment:
Should we keep the `ca.crt` name as for other secrets which have only the CA public key?

Reply:
Agree if it's the only cert in the Secret but this TO Secret contains even private key and cert of the TO.

Reply:
But CA is just one of them. But ok, lets leave it as it is.

Reply:
But we also have the "clients-ca".